src/index.js

```diff
@@ -1,147 +1,53 @@
 'use strict'
 
-const debug = require('debug')
-const log = debug('libp2p:websockets')
-const SW = require('simple-websocket')
-const isNode = require('detect-node')
-let SWS
-if (isNode) {
-  SWS = require('simple-websocket-server')
-} else {
-  SWS = {}
-}
+const connect = require('pull-ws/client')
 const mafmt = require('mafmt')
 const contains = require('lodash.contains')
 const Connection = require('interface-connection').Connection
+const debug = require('debug')
+const log = debug('libp2p:websockets:dialer')
 
-const CLOSE_TIMEOUT = 2000
-// const IPFS_CODE = 421
-
-exports = module.exports = WebSockets
-
-function WebSockets () {
-  if (!(this instanceof WebSockets)) {
-    return new WebSockets()
-  }
+const createListener = require('./listener')
 
-  this.dial = function (ma, options, callback) {
+module.exports = class WebSockets {
+  dial (ma, options, cb) {
     if (typeof options === 'function') {
-      callback = options
+      cb = options
```

Comment:
Let's continue to use `callback` for top level callback and `cb` for internals (like run-series cb)